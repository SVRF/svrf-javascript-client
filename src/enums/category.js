export default {
  FACE_FILTERS: 'Face Filters',
};
