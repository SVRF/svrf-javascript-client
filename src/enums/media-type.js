export default {
  PHOTO: 'photo',
  VIDEO: 'video',
  MODEL_3D: '3d',
};
