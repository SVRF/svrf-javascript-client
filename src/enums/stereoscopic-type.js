export default {
  NONE: 'none',
  TOP_BOTTOM: 'top-bottom',
  LEFT_RIGHT: 'left-right',
};
