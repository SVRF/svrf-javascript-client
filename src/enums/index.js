import category from './category';
import mediaType from './media-type';
import stereoscopicType from './stereoscopic-type';

export default {
  category,
  mediaType,
  stereoscopicType,
};
