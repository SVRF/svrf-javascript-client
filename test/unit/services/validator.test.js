import Validator from '../../../src/services/validator';
import enums from '../../../src/enums';

describe('Validator', () => {
  describe('validateMediaSearchOptions', () => {
    function valid(options) {
      Validator.validateMediaSearchOptions(options);
    }

    function error(options) {
      expect(() => Validator.validateMediaSearchOptions(options))
        .toThrowErrorMatchingSnapshot();
    }

    it('passes if options are not provided', () => valid());
    it('passes for empty options object', () => valid({}));
    it('fails for invalid fields', () => error({sizeeeee: 5}));

    describe('category', () => {
      it('passes for allowed category', () => valid({category: enums.category.FACE_FILTERS}));
      it('fails for invalid category', () => error({category: 'face filterz'}));
    });

    describe('minimumWidth', () => {
      it('passes for valid width', () => valid({minimumWidth: 500}));
    });

    describe('stereoscopicType', () => {
      it('passes for allowed stereotype', () => (
        valid({stereoscopicType: enums.stereoscopicType.TOP_BOTTOM})
      ));
      it('fails for invalid stereotype', () => error({stereoscopicType: 'top down'}));
    });

    describe('type', () => {
      it('passes for allowed media type single value', () => (
        valid({type: enums.mediaType.PHOTO})
      ));
      it('passes for allowed media type multiple value', () => (
        valid({type: [enums.mediaType.PHOTO, enums.mediaType.VIDEO]})
      ));

      it('fails for invalid media type single value', () => error({type: 'photoz'}));
      it('fails for invalid media type multiple value', () => (
        error({type: ['photoz', 'videoz']})
      ));
      it('fails if one of the types is invalid', () => (
        error({type: [enums.mediaType.PHOTO, 'videoz']})
      ));
    });

    describe('pageNum', () => {
      it('passes for valid page number', () => valid({pageNum: 5}));
      it('fails for negative page number', () => error({pageNum: -42}));
    });

    describe('size', () => {
      it('passes for valid page size', () => valid({size: 5}));
      it('fails for 0 page size', () => error({size: 0}));
      it('fails for negative page size', () => error({size: -42}));
      it('fails for large page size', () => error({size: 1000}));
    });
  });
});
