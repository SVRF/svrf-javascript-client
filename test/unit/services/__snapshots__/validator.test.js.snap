// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validator validateMediaSearchOptions category fails for invalid category 1`] = `"\\"Category\\" must be one of [Face Filters]"`;

exports[`Validator validateMediaSearchOptions fails for invalid fields 1`] = `"\\"sizeeeee\\" is not allowed"`;

exports[`Validator validateMediaSearchOptions pageNum fails for negative page number 1`] = `"\\"Page Number\\" must be larger than or equal to 0"`;

exports[`Validator validateMediaSearchOptions size fails for 0 page size 1`] = `"\\"Page Size\\" must be larger than or equal to 1"`;

exports[`Validator validateMediaSearchOptions size fails for large page size 1`] = `"\\"Page Size\\" must be less than or equal to 100"`;

exports[`Validator validateMediaSearchOptions size fails for negative page size 1`] = `"\\"Page Size\\" must be larger than or equal to 1"`;

exports[`Validator validateMediaSearchOptions stereoscopicType fails for invalid stereotype 1`] = `"\\"Stereoscopic Type\\" must be one of [none, top-bottom, left-right]"`;

exports[`Validator validateMediaSearchOptions type fails for invalid media type multiple value 1`] = `"\\"0\\" must be one of [photo, video, 3d]"`;

exports[`Validator validateMediaSearchOptions type fails for invalid media type single value 1`] = `"\\"type\\" must be one of [photo, video, 3d]"`;

exports[`Validator validateMediaSearchOptions type fails if one of the types is invalid 1`] = `"\\"1\\" must be one of [photo, video, 3d]"`;
